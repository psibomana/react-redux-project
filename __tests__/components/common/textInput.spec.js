import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextInput from '../../../src/components/common/textInput';

describe('The Text Input Common Component', () => {
  it('should render as expected', () => {
    const wrapper = shallow(
      <TextInput
        name="Text Input"
        label="Test Text Input"
        value="Default Value"
        onChange= {() => {console.log("Text Input value changed")}}
        error="Error" />
      );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onChange handler', () => {
    const mockOnChange = jest.fn();
    const wrapper = shallow(
      <TextInput
        name="TextInput"
        label="Test Text Input"
        value="Default Value"
        onChange= {mockOnChange}
        error="Error" />
      );
    wrapper.find('input').simulate('change');
    expect(mockOnChange.mock.calls.length).toEqual(1);
  });
});
