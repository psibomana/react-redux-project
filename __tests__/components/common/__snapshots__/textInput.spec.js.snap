// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Text Input Common Component should render as expected 1`] = `
<div
  className="form-group has-error"
>
  <label
    htmlFor="Text Input"
  >
    Test Text Input
  </label>
  <div
    className="field"
  >
    <input
      className="form-control"
      name="Text Input"
      onChange={[Function]}
      type="text"
      value="Default Value"
    />
    <div
      className="alert alert-danger"
    >
      Error
    </div>
  </div>
</div>
`;
